docs/adr/adr-007-incentivized-testnet.md

```diff
@@ -0,0 +1,100 @@
+the network
+
+The issue with the above approach is that there is a possibility that the `daser` would sample over an `ExtendedHeader`
+which it has already sampled. The only way to keep track of which heights have already been sampled is to implement a 
+share cache that would cache the results of sampling processes on disk in such a way that tracks which heights have been
+sampled and whether it was successful or not.
+
+#### [Better error handling](https://github.com/celestiaorg/celestia-node/issues/554)
+The catch-up routine of the `DASer` currently exits the catch-up routine upon error and logs out an instruction to 
+restart the node in order to continue sampling. Instead, there should be retry logic and potentially the graceful 
+shutdown of `DASer` and potentially node.
+
+### Public API 
+The public API is currently being drafted [here](https://github.com/celestiaorg/celestia-node/pull/506/files). There are
+several components that will go into implementing the overall design as well as the individual module implementations.
+
+### [Bad Encoding Fraud Proof](https://github.com/celestiaorg/celestia-node/issues/528) (continuation)
+The full implementation of bad encoding fraud proof (BEFP) generation, propagation and handling is required for the 
+incentivized testnet. At the moment, generation and broadcasting of BEFPs is still in review. 
+
+### Testing TODO LINK TRACKING ISSUE
+One of the biggest priorities for incentivized testnet is to test a lot of the assumptions we are working under in terms
+of sampling in realistic network conditions with a variety of topologies.
+
+Among the most important test cases, the following are required: 
+* TODO
+* TODO 
+* TODO
+
+In terms of testing infrastructure, [testground](https://github.com/testground/testground) will be considered.
+
+### [Wallet Management](https://github.com/celestiaorg/celestia-node/issues/415)
+All DA node types will support sending transactions and querying state-related information, meaning DA nodes 
+will need an account. So far, the standard key utility from cosmos-SDK has been made available via CLI for all node types,
+however more robust key/wallet management via the node's public API is required.
+
+### [`HeaderExchange` extensions and hardening](https://github.com/celestiaorg/celestia-node/issues/497)
+The current implementation of `HeaderExchange` has to be improved and hardened as there is a hard dependency on trusted 
+peers at the moment. Some high-level improvement requirements are: 
+
+* Read/write deadlines on libp2p streams
+* Server-side request limits
+* Server-side maximum request size configuration
+* Adjacency of a batched `ExtendedHeader` response must be verified in `HeaderEx` upon receiving the response
+* Request retries
+* Cycling peers for requests
+
+More detail can be found in the issue linked in the header.
+
+### [IPLD v1 migration + NMT/IPLD optimization](https://github.com/celestiaorg/celestia-node/issues/614)
+@Wondertan todo
```

Comment:
@Wondertan any other issues you can link me to for this?

Reply:
Let's remove IPLD v1 migration in favor of [this](https://github.com/celestiaorg/celestia-node/issues/656). NMT/IPLD storage optimizations can be kept.